api/lib/opentelemetry/trace/tracer.rb

```diff
@@ -0,0 +1,91 @@
+# frozen_string_literal: true
+
+# Copyright 2019 OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    # No-op implementation of Tracer.
+    module Tracer
+      extend self
+
+      CONTEXT_SPAN_KEY = :__span__
+      HTTP_TEXT_FORMAT = nil # TODO: implement HttpTraceContext
+
+      # Formatter for serializing and deserializing a SpanContext into a binary format.
+      module NoopBinaryFormat
+        extend self
+
+        def to_bytes(span_context)
+          raise ArgumentError if span_context.nil?
+
+          []
+        end
+
+        def from_bytes(bytes)
+          raise ArgumentError if bytes.nil?
+
+          SpanContext.INVALID
+        end
+      end
+
+      private_constant(:CONTEXT_SPAN_KEY, :HTTP_TEXT_FORMAT, :NoopBinaryFormat)
+
+      def current_span
+        Context.get(CONTEXT_SPAN_KEY) || Span.INVALID
+      end
+
+      # TODO: This is a helper for the default use-case of extending the current trace with a span.
+      #
+      # The spec-ed API seems a little clunky. Default use-case looks like:
+      # OpenTelemetry.tracer.with_span(OpenTelemetry.tracer.start_span('do-the-thing')) do ... end
```

Comment:
I like this helper 👍 

The spec assumes Java -- looks like in the reference impl, scope implements `Closeable` interface and used along with try-with-resources.. We don't need to follow that pattern.